app/controllers/ncr/proposals_controller.rb

```diff
@@ -29,6 +32,46 @@ def create
       end
     end
 
+    def not_approved
+      cart = Cart.find(params[:id])
+      if cart.approved?
+        redirect_to new_ncr_proposal_path, :alert => "That proposal's already approved. New proposal?"
+      end
+    end
+    def cart_owner
+      cart = Cart.find(params[:id])
+      if cart.requester != current_user
+        redirect_to new_ncr_proposal_path, :alert => 'You cannot restart that proposal'
+      end
+    end
```

Comment:
How about

* Combining these methods and calling it `redirect_if_cart_cant_be_edited` or something?
* Moving them/it to the bottom of the controller?